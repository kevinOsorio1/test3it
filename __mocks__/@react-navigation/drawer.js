module.exports = {
  createDrawerNavigator() {
    return {
      Navigator(props) {
        return props.children;
      },
      Screen() {
        return null;
      },
    };
  },
};
