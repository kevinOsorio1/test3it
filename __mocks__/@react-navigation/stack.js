module.exports = {
  createNativeStackNavigator() {
    return {
      Navigator(props) {
        return props.children;
      },
      Screen() {
        return null;
      },
    };
  },
};
