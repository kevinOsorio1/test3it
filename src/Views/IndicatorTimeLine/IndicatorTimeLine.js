import React from 'react';
import TimeLine from '../../containers/TimeLine/TimeLine';
import {SafeAreaView} from 'react-native';
import useViewHeader from '../../utils/hooks/useViewHeader';
import useIndicatorView from '../../utils/hooks/useIndicatorView';

const IndicatorTimeLine = ({route}) => {
  useViewHeader(route);
  const {indicator} = useIndicatorView(route);
  return (
    <SafeAreaView style={{flex: 1}}>
      <TimeLine prefix={indicator?.prefix} data={indicator?.serie || []} />
    </SafeAreaView>
  );
};

export default IndicatorTimeLine;
