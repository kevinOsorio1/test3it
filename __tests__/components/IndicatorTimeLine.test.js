import React from 'react';
import {render} from '@testing-library/react-native';
import IndicatorTimeLine from '../../src/Views/IndicatorTimeLine/IndicatorTimeLine';
import {NavigationContext} from '@react-navigation/native';
const navContext = {
  isFocused: () => true,
  // addListener returns an unscubscribe function.
  addListener: jest.fn(() => jest.fn()),
};
test('should show indicatorTimeLine', () => {
  const {toJSON} = render(
    <NavigationContext.Provider value={navContext}>
      <IndicatorTimeLine route={{params: {code: 'code', name: 'name'}}} />,
    </NavigationContext.Provider>,
  );
  expect(toJSON).toMatchSnapshot();
});
