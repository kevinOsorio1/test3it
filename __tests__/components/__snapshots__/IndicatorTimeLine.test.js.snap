// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show indicatorTimeLine 1`] = `[Function]`;
